### PR TITLE
Added singletons to documentation

### PR DESCRIPTION
**Pull request by markwardle**
I added singletons to the Container documentation.  I felt that it was lacking this.

### CODE REVIEW

File: singletons.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+permalink: /singletons/
+title: Singletons
+---
+
+# Singletons
+
+Container allows you to register a single instance of a class that will be always be returned when a particular key is requested.  This is known as a singleton and can be registered as a class name, factory closure, or an existing object.
```
Comment (hannesvdvreken) [inline singletons.md]: `be always be` 2 x `be`

File: singletons.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	return $instance;
+});
+
+$firstObject = $container->get('Some\Class');
+$secondObject = $container->get('Some\Class');
+
+$firstObject === $secondObject; // true
+~~~
+
+Note that using the singleton method is identical to using Container's `add` method with an additional, optional third parameter (which defaults to `false`) that marks the key as a singleton.  As such, it is possible to use the same fluid interface to define a singleton as you can when adding non-singleton definitions.
```
Comment (hannesvdvreken) [inline singletons.md]: `as a singleton.  As such` 2 spaces after period.

Comment (hannesvdvreken) [inline singletons.md]: Simpler way of putting it is: `the singleton method is an alias of calling the add method with the third parameter set to true`

**Comment by hannesvdvreken** [comment]
Thanks for helping out on the docs!

**Comment by markwardle** [comment]
No problem.  I made some changes.  I believe the last paragraph is clearer now.

**Comment by hannesvdvreken** [comment]
One last thing you should do: add a menu item to https://github.com/thephpleague/container/blob/gh-pages/_data/menu.yml

**Comment by philipobenito** [comment]
:+1: Happy with this, just addition to the menu and I'm happy to merge, thanks @markwardle and @hannesvdvreken

**Comment by markwardle** [comment]
I added it to the menu.  Wasn't sure where it should go in the order, so I put it after factory closures.

**Comment by markwardle** [comment]
Actually, I moved it to go after auto-dependency-resolution.  Of course, it's up to you, if it even matters.

**Comment by philipobenito** [comment]
:+1:

**Pull request merged**